crates/aranya-client-capi/src/api/defs.rs

```
@@ -1350,69 +1361,445 @@ pub unsafe fn aqc_remove_net_identifier(
     let peer = unsafe { peer.as_underlying() }?;
-    let chan_id = client.rt.block_on(client.inner.aqc().create_bidi_channel(
+
+    let client = client.deref_mut();
+    let chan = client.rt.block_on(client.inner.aqc().create_uni_channel(
         team.into(),
         peer,
         label_id.into(),
     ))?;
-    Ok(AqcBidiChannelId {
-        id: chan_id.aqc_id().into_id().into(),
-    })
+
+    Safe::init(channel, imp::AqcSenderChannel::new(chan));
+    Ok(())
 }
 
+/* TODO(nikki): these will panic currently, so let's stub them
+
 /// Delete a bidirectional AQC channel.
 ///
 /// @param client the Aranya Client [`Client`].
-/// @param chan the AQC channel ID [`AqcBidiChannelId`] of the channel to delete.
+/// @param channel_id the AQC Channel [`AqcBidiChannel`] to delete.
 ///
 /// @relates AranyaClient.
 pub fn aqc_delete_bidi_channel(
     client: &mut Client,
-    chan: &AqcBidiChannelId,
+    channel: &AqcBidiChannelId,
 ) -> Result<(), imp::Error> {
+    // TODO(nikki): change to AqcBidiChannel
     let client = client.deref_mut();
     client
         .rt
-        .block_on(client.inner.aqc().delete_bidi_channel(chan.into()))?;
+        .block_on(client.inner.aqc().delete_bidi_channel(channel.into()))?;
     Ok(())
 }
 
 /// Delete a unidirectional AQC channel.
 ///
 /// @param client the Aranya Client [`Client`].
-/// @param chan the AQC channel ID [`AqcUniChannelId`] of the channel to delete.
+/// @param chan the AQC Channel [`AqcSenderChannel`] to delete.
 ///
 /// @relates AranyaClient.
 pub fn aqc_delete_uni_channel(
     client: &mut Client,
-    chan: &AqcUniChannelId,
+    channel: &AqcSenderChannel,
 ) -> Result<(), imp::Error> {
     let client = client.deref_mut();
     client
         .rt
-        .block_on(client.inner.aqc().delete_uni_channel(chan.into()))?;
+        .block_on(client.inner.aqc().delete_uni_channel(channel.into()))?;
+    Ok(())
+}
+*/
+
+/// Waits until an AQC channel is received from the client.
```

This doc implies that the function will block until an AQC channel is received.
The way the Rust API works is this method returns a channel if it exists, or returns `None` if no channel was received.
In the C API, we should return a specific error if no new channels have been received.

---

That's because it does.
`If there are no messages in the channel's buffer, but the channel has not yet been closed, this method will sleep until a message is sent or the channel is closed. Note that if [close] is called, but there are still outstanding [Permits] from before it was closed, the channel is not considered closed by recv until the permits are released.`

---

Just realized this while testing things out. I don't think we want a blocking API for AQC. This could cause the application to hang unexpectedly. I'd prefer that we implement all the AQC API methods in a non-blocking way.

cc: @benz-spideroak, @djohnson-spideroak, @elagergren-spideroak, @jdygert-spok, @mooreoak